### DISCUSSION TITLE
Listbox with Portal, popper and Transition

### DISCUSSION THREAD

**Discussion by netcelli-tux** [tailwindlabs/headlessui#1690] [Help]
Hi,

I would like to open the Options of a Listbox inside a Portal. This can be useful when using the Listbox component inside a dialog. When using Portal and Transition components with popper, the Options is not properly positioned by popper. 

Example: https://codesandbox.io/s/vigorous-heyrovsky-4cqg8q?file=/src/Select.js

If the Transition component is removed, everything works as expected. Any suggestion how to use Listbox with Portal, Transition and poppe?

Thanks

**Comment by cph0** [comment]
You can move the popper ref, style and attribute props from the options to the transition component. 

Here is a forked example. 

[https://codesandbox.io/embed/kind-kalam-debmgj?codemirror=1](https://codesandbox.io/embed/kind-kalam-debmgj?codemirror=1)
